assets/js/blocks/product-collection/inspector-controls/attributes-control.tsx

```diff
@@ -0,0 +1,74 @@
+/**
+ * External dependencies
+ */
+import { __ } from '@wordpress/i18n';
+import ProductAttributeTermControl from '@woocommerce/editor-components/product-attribute-term-control';
+import { AttributeMetadata } from '@woocommerce/types';
+import { SearchListItem } from '@woocommerce/editor-components/search-list-control/types';
+import { ADMIN_URL } from '@woocommerce/settings';
+import {
+	ExternalLink,
+	// @ts-expect-error Using experimental features
+	// eslint-disable-next-line @wordpress/no-unsafe-wp-apis
+	__experimentalToolsPanelItem as ToolsPanelItem,
+} from '@wordpress/components';
+
+/**
+ * Internal dependencies
+ */
+import { ProductCollectionQuery } from '../types';
+
+const EDIT_ATTRIBUTES_URL = `${ ADMIN_URL }edit.php?post_type=product&page=product_attributes`;
+
+interface AttributesControlProps {
+	woocommerceAttributes?: AttributeMetadata[];
+	setQueryAttribute: ( value: Partial< ProductCollectionQuery > ) => void;
+}
+
+const AttributesControl = ( {
+	woocommerceAttributes,
+	setQueryAttribute,
+}: AttributesControlProps ) => {
+	const selectedAttributes = woocommerceAttributes?.map(
+		( { termId: id } ) => ( {
+			id,
+		} )
+	);
+
+	return (
+		<ToolsPanelItem
+			label={ __( 'Product Attributes', 'woo-gutenberg-products-block' ) }
+			hasValue={ () => !! woocommerceAttributes?.length }
+		>
+			<ProductAttributeTermControl
+				messages={ {
+					search: __( 'Attributes', 'woo-gutenberg-products-block' ),
+				} }
+				selected={ selectedAttributes || [] }
+				onChange={ ( searchListItems: SearchListItem[] ) => {
+					const newValue = searchListItems.map(
+						( { id, value } ) => ( {
+							termId: id as number,
+							taxonomy: value as string,
+						} )
+					);
+
+					setQueryAttribute( {
+						woocommerceAttributes: newValue,
+					} );
+				} }
+				operator={ 'any' }
+				isCompact={ true }
+				type={ 'token' }
+			/>
+			<ExternalLink
+				className="wc-product-collection-panel__external-link"
```

Comment:
This class should probably have the [`wc-block-editor` prefix](https://github.com/woocommerce/woocommerce-blocks/blob/90760e4be19c5e719d8ae20fcb372d01ce63c191/docs/contributors/coding-guidelines.md?plain=1#LL29C6-L29C21).